### ISSUE TITLE
"Only valid json is accepted" problem.

### ISSUE THREAD

**Issue by Inversion-des** [emri99/rockmongo#2]
After updating Mongo and Rockmongo — I have now broken tool.
Every time I try to do something I usually receive " Only valid json is accepted " error, or something [like this](https://drive.google.com/file/d/0Byo4_Oi7CcE7R0tOVk1qajVILVU/preview?pli=1).

Here is my [environment](https://drive.google.com/file/d/0Byo4_Oi7CcE7VmZ0djBYWVdTdFk/preview?pli=1)

Seems like something wrong with the configuration, but I don't know what.
Please advice.

Thank you! (special thank for supporting development of Rockmongo)

**Comment by emri99** [comment]
Internally rockmongo executes command on mongodb server to valid supplied json datas.
Your new environment doesn't allow to execute command as written in the error output:

````
not authorized on metas_test to execute command
````

This issue is "normally" expected when connected to a shard slave, as command is not authorized on slave.
Are you connected to a mongos server ? Check your db rights perhaps...

Note that simple request should work fine in array mode.

**Comment by Inversion-des** [comment]
You are right, problem was with db users rights.

Here is a full story.
After updating Mongo 2.0.6 => 2.6.5, application continued to work, so I did not realize I have to do something more.
In fact I have missed important step ["Upgrade User Authorization Data to 2.6 Format"](http://docs.mongodb.org/master/release-notes/2.6-upgrade-authorization/).
But even after this update I still received same error "not authorized to execute command". 
From the command code I learned that it is an eval command and found [this restriction](http://docs.mongodb.org/manual/reference/method/db.eval/#access-control).

So I had to create special role that grants anyAction on anyResource and assign it to all db users in order to get rid of this error.

And it finally worked. BUT obviously it is not secure and not recommended. And the only way to resolve it properly is to do not use eval in Rockmongo, I guess. Also there is a note "Deprecated since version 3.0"

Thank you for your help! Closing this issue.
